go.mod

```diff
@@ -0,0 +1,118 @@
+module github.com/openshift/installer
+
+go 1.12
+
+require (
+	cloud.google.com/go/bigtable v1.1.0 // indirect
+	github.com/Azure/azure-sdk-for-go v36.2.0+incompatible
+	github.com/Azure/go-autorest/autorest v0.9.2
+	github.com/Azure/go-autorest/autorest/adal v0.8.1-0.20191028180845-3492b2aff503
+	github.com/Azure/go-autorest/autorest/azure/auth v0.4.0
+	github.com/Azure/go-autorest/autorest/to v0.3.0
+	github.com/Netflix/go-expect v0.0.0-20180615182759-c93bf25de8e8
+	github.com/apparentlymart/go-cidr v1.0.1
+	github.com/awalterschulze/gographviz v0.0.0-20170410065617-c84395e536e1
+	github.com/aws/aws-sdk-go v1.25.3
+	github.com/containers/image v3.0.2+incompatible
+	github.com/coreos/ignition v0.28.0
+	github.com/dmacvicar/terraform-provider-libvirt v0.6.0
+	github.com/dustinkirkland/golang-petname v0.0.0-20191129215211-8e5a1ed0cff0 // indirect
+	github.com/ghodss/yaml v1.0.0
+	github.com/golang/mock v1.3.1
+	github.com/gophercloud/gophercloud v0.6.1-0.20191019020556-0907b320e0ac
+	github.com/gophercloud/utils v0.0.0-20191020172814-bd86af96d544
+	github.com/hashicorp/go-azure-helpers v0.10.0
+	github.com/hashicorp/go-plugin v1.0.1
+	github.com/hashicorp/logutils v1.0.0
+	github.com/hashicorp/terraform v0.12.17
+	github.com/hashicorp/terraform-plugin-sdk v1.3.0
+	github.com/hinshun/vt10x v0.0.0-20180616224451-1954e6464174
+	github.com/kr/fs v0.1.0 // indirect
+	github.com/libvirt/libvirt-go v5.0.0+incompatible
+	github.com/metal3-io/baremetal-operator v0.0.0
+	github.com/metal3-io/cluster-api-provider-baremetal v0.0.0
```

Comment:
is it possible to collect these replacement only imports out of the normal group or go mod doesn't allow that?

Reply:
I'm not sure I follow, can you elaborate?

Reply:
there are a bunch of these 0.0.0 or 0.0.0-000001100010-00000
so I was curious why these are such and if there is special about all these separating out and collecting them in a separate group in require section was useful?

Reply:
those are replaced by the pinned version in the replace directive. `go mod tidy` sorts all deps automatically, so manual ordering can't really be done here, and I doubt we can split them into multiple groups in one go.mod file

Reply:
If it matters, all of the `v0.0.0-00010101000000-000000000000` tags can be replaced with `v0.0.0` for readability. Otherwise, having the actual dependency version in the replace directive section is typical. See https://github.com/golang/go/wiki/Modules#when-should-i-use-the-replace-directive.

Reply:
They got pulled with that long zero version automatically from the respective dep (the replaces however are not added automatically in `go mod init`). I shortened them all now to make it easier on the eyes.

Note that it's not necessary for replaced deps to have a zero version, tho. I think it's a good practice though to give v0.0.0 in order to signal they are being replaced down below.